lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PuppeteerSharp.Tests.PageTests.Events
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class CloseTests : PuppeteerPageBaseTest
+    {
+        public CloseTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldWorkWithWindowClose()
+        {
+            var newPageTaskSource = new TaskCompletionSource<Page>();
+            Browser.TargetCreated += async (sender, e) =>
+            {
+                if (!newPageTaskSource.Task.IsCompleted)
```

Comment:
can't we just use `TrySetResult`, this way only the first call will succeed

Reply:
Why didn't you teach me about `TrySetResult` before? :joy:

Reply:
I was sure we used it somewhere in the code...